openjdk/openjdk18-image-stream.json

```diff
@@ -95,6 +95,59 @@
                     }
                 ]
             }
+        },
+        {
+            "kind": "ImageStream",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "java",
+                "annotations": {
+                    "openshift.io/display-name": "Red Hat OpenJDK 8",
+                    "openshift.io/provider-display-name": "Red Hat, Inc.",
+                    "version": "1.4.11"
+                }
+            },
+            "labels": {
+                "xpaas": "1.4.11"
+            },
+            "spec": {
+                "tags": [
+                    {
+                        "name": "8.0",
+                        "annotations": {
+                            "openshift.io/display-name": "Red Hat OpenJDK 8",
+                            "description": "Build and run Java applications using Maven and OpenJDK 8.",
+                            "iconClass": "icon-rh-openjdk",
+                            "tags": "builder,java,openjdk",
+                            "supports": "java:8",
+                            "sampleRepo": "https://github.com/jboss-openshift/openshift-quickstarts",
+                            "sampleContextDir": "undertow-servlet",
+                            "version": "8.0"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:1.3"
```

Comment:
Can we just use `:latest` here instead?  Otherwise, we need to update the image stream definition every time we release an update to the Java image, e.g. 1.4.  I don't know if the latest tag is applied to the images in RHCC though.

Reply:
That only solve (less than) half the problem, one would need to do:

`oc new-app redhat-openjdk18-openshift:latest .`

Which is an improvement but still hard to type, would need to search and copy-paste.

Reply:
sorry, what i meant was use latest for the image reference, i.e. `registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift:latest`.  I'm totally on board with java:8.0 (or maybe just java:8)

Reply:
Ah, cool...I can update that.

I was struggling with the ".0" part as well, I was trying to be consistent. I'm seeing the node.js image stream from SCL uses just whole number "8" but saw php had "7.0" and "7.1". I guess if one used just "7" in the php case, would it follow the latest "7.x" releases or would it be "7.0".

Trying to come up with a good convention across the board.

/cc @bparees

Reply:
I think the issue is that for Java, there is no 8.x, it's just Java 8, 9, 10, 11, etc.  As for `1.3` vs `latest`, those tags really only represent the version of the integration scripts within the image.  They may relate to a specific Java update, but that would be updated with every image build anyway (i.e. yum install java-1.8.0-openjdk) and would not be reflected in a new tag (e.g. CVE updates).  If there were a new version, there would be a new image name, e.g. openjdk19-openshift.  HTH

Reply:
the pattern we've established elsewhere is that you would not version your image due to changing your scripts, the script changes should be compatible.

i'm not going to claim that hasn't caused us some pain, though.